### PR TITLE
Fix tob-7

### PR DESCRIPTION
**Pull request by 42bchen**
https://www.notion.so/perp/Full-Review-6a7588070f9949658fb3ec8eaab954b2

### CODE REVIEW

File: contracts/twap/CumulativeTwap.sol [hunk 1]
```diff
@@ -101,8 +101,8 @@ contract CumulativeTwap is BlockContext {
         }
         // case3. in the middle
         else {
-            uint256 observationTimeDelta = atOrAfter.timestamp - beforeOrAt.timestamp;
-            uint256 targetTimeDelta = targetTimestamp - beforeOrAt.timestamp;
+            uint256 observationTimeDelta = atOrAfter.timestamp.sub(beforeOrAt.timestamp);
```
Comment (42bchen) [inline contracts/twap/CumulativeTwap.sol:104]: we check underflow on `CumulativeTwap:157`, but still use safeMath to handle math operation

File: contracts/twap/CumulativeTwap.sol [hunk 2]
```diff
@@ -49,9 +49,9 @@ contract CumulativeTwap is BlockContext {
         // so 255 + 1 will be 0
         currentObservationIndex++;
 
-        uint256 elapsedTime = lastUpdatedTimestamp - lastObservation.timestamp;
+        uint256 elapsedTime = lastUpdatedTimestamp.sub(lastObservation.timestamp);
         observations[currentObservationIndex] = Observation({
-            priceCumulative: lastObservation.priceCumulative + (lastObservation.price * elapsedTime),
+            priceCumulative: lastObservation.priceCumulative.add(lastObservation.price.mul(elapsedTime)),
```
Comment (42bchen) [inline contracts/twap/CumulativeTwap.sol:54]: `tob-7`

**Comment by KimiWu123** [comment]
could you add some description or copy from the report in this PR?

**Review by KimiWu123 — APPROVED** [review]

**Comment by tienshaoku** [comment]
Closing this. Go to https://github.com/perpetual-protocol/perp-oracle/pull/20 instead!

**Pull request closed without merging**